addons/binding/org.openhab.binding.tplinksmarthome/README.md

```diff
@@ -129,59 +140,68 @@ via openHAB activates the switch directly.
 
 ## Prerequisites
 
-Before using Smart Plugs with openHAB the devices must be connected to the Wi-Fi network. This can be done using the
-TP-Link provided mobile app Kasa.
+Before using Smart Plugs with openHAB the devices must be connected to the Wi-Fi network.
+This can be done using the TP-Link provided mobile app Kasa.
 
 ## Discovery
 
-Devices can be auto discovered in the same local network as the openHAB application. It's possible to connect to
-devices in a different network, but these must be added manually.
+Devices can be auto discovered in the same local network as the openHAB application.
+It's possible to connect to devices on a different network, but these must be added manually by ipAddress.
+It's not possible to connect to devices on a different network using `deviceId` as configuration.
 
 ## Thing Configuration
 
-The thing id is the product type in lower case. For example `HS100` has id `hs100`.
+The thingId is the product type in lower case. For example `HS100` has thingId `hs100`.
 
-The thing has a few configuration parameters:
+The thing can be configured or by `ipAddress` or by `deviceId`.
+If the one of them is used the other is automatically set by the binding.
+When manually configured it's preferred to set the `deviceId` because if the ip address of the device would change this will be automatically updated.
+When setting the `deviceId` it can take up to 1 minute before the ip address is set. By manually triggering a discovery process it will update it faster.
```

Comment:
@kaikreuzer @sjka, is this okay? I remembered this comment: https://github.com/openhab/openhab2-addons/issues/2916#issue-279370881

Reply:
Well, that's the to-be-accepted drawback of that kind of configuration then.

Reply:
I've updated the documentation to better state it's dependency on the discovery service. It works @kaikreuzer mentions. it's only used to set the ip address in case the user doesn't set the ip address or in case there the ip address of the device changed. It doesn't depend on discovery for status or controlling the device.
I thought about some more fancy code by triggering the discovery when no ip address is know. This would make it independent on (background) discovery, but would require the binding handling timeout, correctly stopping and some more checks to make it clean and efficient. But that seem to be to much for the gains it would get and make the code much more complex (and probably spending a lot of time to get it correct)